### PR TITLE
Add a `<Header />`, `<AppTitle />` and `<AI2Logo />` component.

### PR DESCRIPTION
**Pull request by codeviking**
This adds components that support rendering a standard application
header.

Part of this required modifying the default theme. I added colors
per the AI2 website and deleted the `testColor` mechanism we
were using to prove out that approach.

### CODE REVIEW

**Comment by codeviking** [comment]
@jonborchardt Here you go!

File: src/lib/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { AppBar, Grid } from '@material-ui/core'
+import styled from 'styled-components';
+
+interface Props {
+    position?: 'fixed' | 'absolute' | 'sticky' | 'static' | 'relative',
```
Comment (jonborchardt) [inline src/lib/components/Header.tsx]: move to types?
type Position = 'fixed' | 'absolute' | 'sticky' | 'static' | 'relative';
then
interface Props {
    position?: Position

File: src/lib/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * applications.
+ */
+export const Header = (props: Props) => {
+    const { children, position } = props;
+    // Make sure the children are an array, as we allow them to specify
+    // a single descendent.
+    const childArray = Array.isArray(children) ? children : [ children ];
+    return (
+        <StyledAppBar position={position}>
+            <Grid container spacing={24} alignItems="center" wrap="wrap">
```
Comment (jonborchardt) [inline src/lib/components/Header.tsx]: we could expose more of these... but not needed

File: src/lib/components/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+const StyledAppBar = styled(AppBar)`
+    border-top: 4px solid ${props => props.theme.palette.highlight.main};
+    padding: 1rem 2rem;
+    && {
+        background: #fff;
+    }
+`
+
```
Comment (jonborchardt) [inline src/lib/components/Header.tsx]: remove extra lines

File: src/lib/components/Header.tsx (hunk 1 above)
Comment (codeviking) [reply src/lib/components/Header.tsx]: 👍 I wonder if I can extract the type from their definition. I'll dig in and see what I can do.

File: src/lib/components/Header.tsx (hunk 2 above)
Comment (codeviking) [reply src/lib/components/Header.tsx]: You mean as as a separate component, i.e. a `<FlexRow />` or something like that?

File: src/lib/components/Header.tsx (hunk 3 above)
Comment (codeviking) [reply src/lib/components/Header.tsx]: 👍

File: src/lib/components/Header.tsx (hunk 2 above)
Comment (jonborchardt) [reply src/lib/components/Header.tsx]: no, i mean we could allow user to pass in more props.... wrap, spacing, and just set these as defaults

Comment (codeviking) [reply src/lib/components/Header.tsx]: Ah! Totally. I figured I'd leave those out until we need them -- as maybe people won't need them at all!

**Pull request merged**